### ISSUE TITLE
install python on stntest and toad servers

### ISSUE THREAD

**Issue by troddick** [USGS-WiM/STNWeb#392]
Need Python27 installed on stntest and toad to allow for scripts to be run for sensor data files. 
I attempted to do it myself, but I am getting:

 'AttributeError: 'module' object has no attribute 'openssl_md_meth_names'

I installed python 2.7, copied over the contents of my Lib folder to stntest and replaced the Lib folder in Python27 with it in hopes that would bring all the files I need. 
In order for the stn_script.py to run successfully, the server needs to have the following installed:
(from my notes after Greg helped me install on my machine)

pip install pandas
pip install matplotlib
pip install pytz
pip install netCDF4

http://www.lfd.uci.edu/~gohlke/pythonlibs/
 (move whl file to the scripts folder then run it)
pip install scipy ...whl file name

add numpy (same as scipy)
pip install numpy.... whl file name

Tonia

**Comment by aaronstephenson** [comment]
So, what are you asking for? Help with that error?

**Comment by aaronstephenson** [comment]
When installing python libs through a method other than pip, you really need to ensure that the versions match. I strongly suspect that the whl files you downloaded from gohlke's site don't match the version of python you installed (and yes, the minor and maintenance numbers of the version matter).

**Comment by troddick** [comment]
I am asking that Python vs 2.7 be installed on the 2 servers. (STNTest and Toad)

**Comment by aaronstephenson** [comment]
But you say in your original message that you installed python 2.7. So could you please clarify your request?

**Comment by troddick** [comment]
OK. So, as you can probably assume,  I have zero experience or knowledge of working with Python, so I am assuming I did something wrong. Jeremy looked at it with me and was unable to figure it out either.
I did install python27.

The folder was created in the C: but it did not have a Script folder. I copy/pasted my local machine's Script folder over. I also copy/pasted (replaced) the Lib folder from my machine. 

I am not able to do any installing of the required files, or get access to using pip in command line.
I am thinking maybe the easiest solution is to let you install the requirements on the servers, as I probably should have requested to begin with. 

So, I can delete the Python27 folder, if that helps, or you could go in and see what I'm missing or did wrong.

**Comment by aaronstephenson** [comment]
I uninstalled the python binaries you installed, then installed everything from scratch. It all installed just fine, but I don't know what else you need done. Is there a way for me to test your scripts, or do you just want to take it from here?

(side note: You are in the Administrators group, so you should have full permission to do installs and uninstalls. Also note that in older versions of python 2.7, pip was not included by default, so you have to install it manually, and it's also not registered with the OS, so the only way to access it easily is to call python first, e.g. `python -m pip ...`)

**Comment by troddick** [comment]
Thank you!
I moved the Assets folder over to stntest test server in the STNServices2 folder. There are a bunch of scripts in there that all get used from stn_script.py. I ran a test to see if it would work and it does not. 

Looks like some imports are needed. So far, numpy. I went to the link Greg sent me (http://www.lfd.uci.edu/~gohlke/pythonlibs/), installed the numpy-1.13.1...., and moved it into the Scripts folder. I tried doing pip install that file name and I'm getting an error.
![image](https://user-images.githubusercontent.com/1580076/30127690-bbd16cb6-9305-11e7-828a-e1591872e2c7.png)

Here is the Command Prompt code I ran to get the script going: 
C:\Python27\python C:\inetpub\wwwroot\STNServices2\Assets\Scripts\stn_script.py 
"C:\inetpub\wwwroot\STNServices2\Assets\Scripts\documentation\NCCAR00007_1511451_sea.csv" 
"C:\inetpub\wwwroot\STNServices2\Assets\Scripts\documentation\NCCAR12248_9983816_air.csv" 
"C:\stntemp\stormtide\16025INSTRUMENT7649DFID371909052017142609\NCCAR00007_1511451_sea_output" "Tonia Roddick" "troddick@usgs.gov" "www.usgs.gov" "Measurement Specialties" "Hobo" 16024 16023 7649 7650 44.465456 44.45646 -88.546546 -90.5465 UTC False NAVD88 2.4 1 2.3 2 "Saline Water" 2.6 2.4 "20170823 1454" "20170823 1455" "Testtest" "20161008 0800" "20161008 0800" "20161010 0800" "20161010 0800" True

**Comment by aaronstephenson** [comment]
Numpy is already installed. You can confirm this by issuing this command in the terminal: `python -c "import numpy"`.

Also, like I mentioned previously, in version 2.7 you need to call pip thru python, e.g. `python -m pip install ...`.

I just ran `python C:\inetpub\wwwroot\STNServices2\Assets\Scripts\stn_script.py` and got the error "Import Error: cannot import name NUMPY_MKL", which means a module or submodule is missing. I will look into it after my next meeting.

**Comment by troddick** [comment]
Thank you

**Comment by aaronstephenson** [comment]
OK, I got the stn_script.py to run without errors. I didn't include any arguments, tho, so it didn't get very far. Could you give it a try @troddick ?

Turns out we need to install the numpy from whl _first_, before installing pandas, because numpy is a prerequisite for pandas and it will be installed from PyPI without the mkl extension if it isn't already installed.

**Comment by troddick** [comment]
The script runs successfully when pointing the output location (where output files will be written) to the same location as the scripts. We would like to have a separate temporary directory location instead (that will be created upon start and deleted upon completion via the services). When attempting this (after creating the location where it should be stored manually), I get the below error.
![image](https://user-images.githubusercontent.com/1580076/30276477-97ca9888-96ca-11e7-90cf-d46dae3e0d3b.png)

So, in summary, the python install with all of it's requirements was a success (we will need this on Toad), but I think we need some write privilege to allow the created output files to be stored temporarily somewhere on the server. I don't care where. I can update the services to create the directory in any location you see fit.

**Comment by aaronstephenson** [comment]
Ah yes, that error is most likely because the C drive is just protected by default, so Python wouldn't have permission to alter anything there. In production, you should put that folder on a D drive or something else. I would recommend D:\stn\tmp.

In the meantime, on the test server, you can launch CMD as Administrator and you should have the necessary permissions. Or, your service could be given elevated permissions (if it's a Windows service). Or, if it's a web service, we could give elevated permissions to whichever local service runs the web service process (either w3wp or svchost, most likely the former).

**Comment by aaronstephenson** [comment]
OK @troddick I have installed Python 2.7 64-bit and the requested packages on Toad. I also created a folder at D:\stn\tmp for use by your script. Could you give it a try and let me know how it goes?

**Comment by troddick** [comment]
I will try this and get back you by the end of day or tomorrow morning. (on hospital wifi right now)
Thanks!

**Comment by troddick** [comment]
Running the commander as admin worked on stntest.
I moved the Scripts contents to toad within the temp folder as a way to test the script, and it works! (Hurray!) 
To test via the stnservices and the stnweb client app on stntest, if you could give the elevated permissions for the web service process. It is w3wp, because I googled that and it says: "An Internet Information Services (IIS) worker process is a windows process (w3wp.exe) which runs Web applications, and is responsible for handling requests sent to a Web Server for a specific application pool. " and stnservices is running through IIS. :)

**Comment by troddick** [comment]
Aaron, I can do this (give elevated permissions for the web service process on stntest) if you are busy with other things. I did a quick search and was not able to find a clear solution as to how to do it, but if it's easy enough, just tell me what to do and I can do it.

**Comment by aaronstephenson** [comment]
I gave full permission to C:\stntemp the user "Local Service". Give it a try.

**Comment by troddick** [comment]
The folders are being creating at that location. Thank you! The script is failing after the files are moved from s3 to the temp folder, but I'm assuming that something in my code. Main objective is successful though. 
Thanks!
